kernel: bump 5.10 to 5.10.100

graysky2 opened the pull request:
All patches automatically rebased.

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <graysky@archlinux.us>

graysky2 committed:
kernel: bump 5.10 to 5.10.98

Manually rebased:
	bcm27xx/patches-5.10/950-0675-drm-vc4-hdmi-Drop-devm-interrupt-handler-for-CEC-int.patch

All other patches automatically rebased.

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <graysky@archlinux.us>

graysky2 commented:
@rsalvaterra - Please check my rebase of that patch when you have a minute.

Borromini commented:
```
Build-tested: realtek/generic
Run-tested: realtek/generic
```

Signed-off-by: Stijn Segers <foss@volatileystems.org>

PerkelSimon commented:
I applied the patch with success but still on 5.10.96 after rebuild, even if the kernel 5.10 is @5.10.98  

![kernel](https://user-images.githubusercontent.com/79952505/152716469-7299affa-f87e-4370-a7a0-ce7c3697d015.png)

rsalvaterra commented:
> @rsalvaterra - Please check my rebase of that patch when you have a minute.

Looks sane to me. If your Pi 4 didn't complain, it should be fine.

graysky2 commented:
Guys - I got a build error now whereas it built fine when I opened this PR.  Please do not merge yet.

graysky2 committed:
kernel: bump 5.10 to 5.10.99

Had to update generic defconfig (make kernel_menuconfig CONFIG_TARGET=generic)
for this bump, but since that only modifies the target defined in .config,
and since that target also needed to be updated for unrelated reasons, manually
propagated the newly added symbol to the generic config.

Removed upstreamed:
    pending-5.10/860-Revert-ASoC-mediatek-Check-for-error-clk-pointer.patch[1]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.10.99&id=080f371d984e8039c66db87f3c54804b0d172329

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <graysky@archlinux.us>

rsalvaterra commented on target/linux/bcm27xx/bcm2711/config-5.10:
```diff
@@ -188,6 +186,7 @@ CONFIG_FONT_8x8=y
 CONFIG_FONT_SUPPORT=y
 CONFIG_FRAMEBUFFER_CONSOLE=y
 # CONFIG_FRAMEBUFFER_CONSOLE_DETECT_PRIMARY is not set
+# CONFIG_FRAMEBUFFER_CONSOLE_LEGACY_ACCELERATION is not set
```
Hm. I'm pretty sure all `CONFIG_FRAMEBUFFER_CONSOLE=y` targets will complain about this missing symbol. Since there are quite a few of them…
```
rui@crystalwell:~/Sources/OpenWrt/openwrt/target$ find . -type f -name "config-5.10" -exec grep CONFIG_FRAMEBUFFER_CONSOLE=y {} +
./linux/bcm27xx/bcm2711/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
./linux/bcm27xx/bcm2710/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
./linux/bcm27xx/bcm2708/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
./linux/bcm27xx/bcm2709/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
./linux/sunxi/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
./linux/mediatek/mt7623/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
./linux/x86/legacy/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
./linux/x86/64/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
./linux/x86/generic/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
./linux/layerscape/armv8_64b/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
./linux/gemini/config-5.10:CONFIG_FRAMEBUFFER_CONSOLE=y
rui@crystalwell:~/Sources/OpenWrt/openwrt/target$ 
```
… it's a good idea to move this symbol to the generic config.

rsalvaterra commented:
Apart from the config changes (which I'd prefer to be a separate commit, to be honest), the 5.10.99 bump matches mine.

graysky2 commented:
Good idea... with respect to the config changes:

1. If in a separate commit, it would need to be merged first as the build will error out without them.
2. What is the make target to update the generic config rather a specific target?

jclong83 commented:
I think what @rsalvaterra means is a separate commit, can be included in this PR. [atomic commits](https://sparkbox.com/foundry/atomic_commits_with_git)

ldir-EDB0 commented:
I'm with graysky - the config change should be in the same commit as the kernel version bump - otherwise bisect won't work.  It's how we usually do it.

rsalvaterra commented:
> 1. If in a separate commit, it would need to be merged first as the build will error out without them.  I thought in the past, if the kernel bump required these, it was ok to include them in that rebase PR.

I meant a commit just adding `# CONFIG_FRAMEBUFFER_CONSOLE_LEGACY_ACCELERATION is not set` to the generic config, between the 5.10.98 and the 5.10.99 kernel bumps, as part of this pull request, of course. But like @ldir-EDB0 said, that's probably excessive (see below).

> 2. What is the make target to update the generic config rather a specific target?

AFAIK, I don't think there is one. I manually edit the generic config when needed (and sort the symbols afterwards, to make sure).

> I'm with graysky - the config change should be in the same commit as the kernel version bump - otherwise bisect won't work. It's how we usually do it.

I agree that adding the `CONFIG_FRAMEBUFFER_CONSOLE_LEGACY_ACCELERATION` symbol to the generic config can be part of the 5.10.99 kernel bump, but the removal of the unneeded bcm2711 symbols (which happened as a consequence of a `make kernel_oldconfig CONFIG_TARGET=subtarget`, most likely) is unrelated. The whole tree is desperately needing a config symbol cleanup, but that's another story. 😉

graysky2 commented:
Thanks for the discussion, all.  Confirmed that this new commit builds, boots, and runs.

> The whole tree is desperately needing a config symbol cleanup, but that's another story.

Someone should write a script that does this if we don't already have one, no?

rsalvaterra commented:
> Someone should write a script that does this if we don't already have one, no? Seems like something the botbuild farm could do since they have the toolchains.

I wish it were scriptable, but alas… there's still manual labour involved, unfortunately. It _might_ be automated once the initial cleanup is done, but I can't be certain, as I only had a cursory look.

graysky2 commented:
Yeah, I don't see a make target that just updates with the file without launching an interactive config editor.  In principal, a script could loop through the targets/subtargets and run whatever make target does that non-interactively, and commit if changed.

graysky2 committed:
kernel: bump 5.10 to 5.10.100

All patches automatically rebased.

Build system: x86_64
Build-tested: bcm2711/RPi4B, mt7622/RT3200
Run-tested: bcm2711/RPi4B, mt7622/RT3200

Signed-off-by: John Audia <graysky@archlinux.us>

hauke commented:
Thank you for your patches, I applied them to master in e9c1c83679891a02e3b95b954847a95621cbbc79, 5e43dd1fa7ed294a4ba7a3aa32c8960d9f3818e7 and c391dcdf86b3001b98725225b54ae60f4b6d493c
